src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -1245,8 +1245,32 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         llvm::Value* const_list = builder->CreateAlloca(const_list_type, nullptr, "const_list");
         list_api->list_init(type_code, const_list, *module, x.n_args, x.n_args);
         int64_t ptr_loads_copy = ptr_loads;
-        ptr_loads = 1;
         for( size_t i = 0; i < x.n_args; i++ ) {
+            if (ASR::is_a<ASR::CPtr_t>(*ASRUtils::expr_type(x.m_args[i]))) {
+                bool is_argument = false;
+                ASR::expr_t *var = x.m_args[i];
+                if (is_a<ASR::Var_t>(*var)) {
+                    ASR::symbol_t *var_sym = ASR::down_cast<ASR::Var_t>(var)
+                        ->m_v;
+                    if (is_a<ASR::Variable_t>(*var_sym)) {
+                        ASR::Variable_t *v = down_cast<ASR::Variable_t>(var_sym);
+                        if (v->m_intent == intent_local ||
+                            v->m_intent == intent_return_var ||
+                            !v->m_intent) {
+                            is_argument = false;
+                        } else {
+                            is_argument = true;
+                        }
+                    }
+                }
```

Comment:
This whole code here should be refactored into a function that should be put into ASRUtils. We might already have a function for this.